app/src/main/java/com/epmedu/animeal/MainActivity.kt

```diff
@@ -40,4 +45,18 @@ class MainActivity : ComponentActivity() {
             }
         }
     }
+
+    override fun signInWithFacebook(
+        onSuccess: () -> Unit,
+        onError: (Throwable) -> Unit
+    ) {
+        Amplify.Auth.signInWithSocialWebUI(
+            AuthProvider.facebook(),
+            this,
+            { signInResult ->
+                if (signInResult.isSignInComplete) onSuccess()
+            },
+            {}
```

Comment:
Even if we are not passing anything meaningful as onError from oboarding, I guess we can just pass that lambda further to Amplify.Auth here

Reply:
Done